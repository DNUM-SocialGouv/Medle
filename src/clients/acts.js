import { saveAs } from "file-saver"
import fetch from "isomorphic-unfetch"

import { ACTS_ENDPOINT, API_URL } from "../config"
import { handleAPIResponse } from "../utils/errors"
import { METHOD_DELETE, METHOD_POST, METHOD_PUT } from "../utils/http"

export const findAct = async ({ id, headers }) => {
  const response = await fetch(API_URL + ACTS_ENDPOINT + "/" + id, { headers })
  return handleAPIResponse(response)
}

export const searchActsByKey = async ({ key, value, headers }) => {
  const response = await fetch(`${API_URL}${ACTS_ENDPOINT}?${key}=${value}`, {
    headers: { ...headers, "Content-Type": "application/json" },
  })

  const { elements } = await handleAPIResponse(response)
  return elements
}

const buildQueryParams = (params) => {
  const { search, startDate, endDate, hospitals, profiles, asker, requestedPage } = params

  const arr = []
  if (search) {
    arr.push(`fuzzy=${search}`)
  }
  if (startDate) {
    arr.push(`startDate=${startDate}`)
  }
  if (endDate) {
    arr.push(`endDate=${endDate}`)
  }
  if (hospitals?.length) {
    arr.push(`hospitals=${hospitals.map((elt) => elt.value).join(",")}`)
  }
  if (profiles?.length) {
    arr.push(`profiles=${profiles.map((elt) => elt.value).join(",")}`)
  }
  if (asker?.value) {
    arr.push(`asker=${asker?.value}`)
  }
  if (requestedPage) {
    arr.push(`requestedPage=${requestedPage}`)
  }
  return arr.length ? "?" + arr.join("&") : ""
}

export const searchActs = async (params) => {
  const { headers } = params

  const queryParams = buildQueryParams(params)

  const response = await fetch(`${API_URL}${ACTS_ENDPOINT}${queryParams}`, { headers })

  return handleAPIResponse(response)
}

export const createAct = async ({ act, headers }) => {
  const response = await fetch(API_URL + ACTS_ENDPOINT, {
    method: METHOD_POST,
    headers: { ...headers, "Content-Type": "application/json" },
    body: JSON.stringify(act),
  })
  return handleAPIResponse(response)
}

export const updateAct = async ({ act, headers }) => {
  const response = await fetch(API_URL + ACTS_ENDPOINT + "/" + act.id, {
    method: METHOD_PUT,
    headers: { ...headers, "Content-Type": "application/json" },
    body: JSON.stringify(act),
  })
  return handleAPIResponse(response)
}

export const deleteAct = async ({ id, headers }) => {
  const response = await fetch(`${API_URL}${ACTS_ENDPOINT}/${id}`, { method: METHOD_DELETE, headers })
  return handleAPIResponse(response)
}

export const fetchExport = async (params) => {
  const { headers } = params

  const queryParams = buildQueryParams(params)

  saveAs(`${API_URL}${ACTS_ENDPOINT}/export${queryParams}`, { headers })
}
