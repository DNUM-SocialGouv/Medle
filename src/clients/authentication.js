import fetch from "isomorphic-unfetch"

import { API_URL, LOGIN_ENDPOINT } from "../config"
import { handleAPIResponse2 } from "../utils/errors"
import { METHOD_POST } from "../utils/http"

export const authenticate = async (email, password) => {
  const response = await fetch(API_URL + LOGIN_ENDPOINT, {
    method: METHOD_POST,
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ email, password }),
  })

  const tokenCookie = response.headers.get("Set-Cookie") // may be null in CSR mode or because of CORS or secure cookie. OK in SSR and good for us for testing purpose.
  
  const { user, token } = await handleAPIResponse2(response)
  
  if (!user) {
    throw new Error("Authentication failed")
  }
  
  return {
    headers: {
      cookie: tokenCookie,
    },
    user,
    token
  }
}
