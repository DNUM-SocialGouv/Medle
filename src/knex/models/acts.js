import moment from "moment"

export const buildActsFromDB = acts =>
   acts.map(elt => ({
      id: elt.id,
      createdAt: elt.created_at,
      updatedAt: elt.updated_at,
      deletedAt: elt.deleted_at,
      casesFk: elt.cases_fk,
      pvNumber: elt.pv_number,
      examinationDate: elt.examination_date,
      asker: elt.asker,
      addedBy: elt.added_by,
      data: elt.data,
   }))

// from js to db
export const buildActFromJSON = data => {
   const res = { extra_data: {} }

   const mainKeys = {
      id: "id",
      internalNumber: "internal_number",
      pvNumber: "pv_number",
      examinationDate: "examination_date",
      profile: "profile",
      askerId: "asker_id",
      addedBy: "added_by",
      hospitalId: "hospital_id",
   }

   Object.keys(data).forEach(key => {
      if (mainKeys[key]) {
         res[mainKeys[key]] = data[key]
      } else {
         res.extra_data[key] = data[key]
      }
   })

   return res
}

// from db to js
export const buildActFromDB = data => ({
   id: data.id,
   internalNumber: data.internal_number,
   pvNumber: data.pv_number,
   examinationDate: moment(data.examination_date).format("YYYY-MM-DD"),
   profile: data.profile,
   asker: { id: data.asker_id, name: data.asker_name },
   user: {
      id: data.added_by,
      firstName: data.user_first_name,
      lastName: data.user_last_name,
      email: data.user_email,
   },
   hospital: { id: data.hospital_id, name: data.hospital_name },
   ...data.extra_data,
})
