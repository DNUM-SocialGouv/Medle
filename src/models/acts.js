import moment from "moment"

import { ISO_DATE } from "../utils/date"

// from Knex entity to model (JS) entity
export const transform = (knexData) => {
  return !knexData
    ? null
    : {
        id: knexData.id,
        internalNumber: knexData.internal_number,
        pvNumber: knexData.pv_number,
        examinationDate: moment(knexData.examination_date).format(ISO_DATE),
        profile: knexData.profile,
        asker: { id: knexData.asker_id, name: knexData.asker_name },
        user: {
          id: knexData.added_by,
          firstName: knexData.user_first_name,
          lastName: knexData.user_last_name,
          email: knexData.user_email,
        },
        hospital: { id: knexData.hospital_id, name: knexData.hospital_name },
        ...knexData.extra_data,
      }
}
// from Knex entity to model (JS) entity for export needs
export const transformForExport = (knexData) => {
  return !knexData
    ? null
    : {
        id: knexData.id,
        internalNumber: knexData.internal_number,
        pvNumber: knexData.pv_number,
        examinationDate: moment(knexData.examination_date).format(ISO_DATE),
        profile: knexData.profile,
        asker: knexData.asker_name,
        user: knexData.user_email,
        hospital: knexData.hospital_name,
        ...knexData.extra_data,
      }
}

export const transformAll = (list) => list.map((knexData) => transform(knexData))
export const transformAllForExport = (list) => list.map((knexData) => transformForExport(knexData))

// from model (JS) entity to Knex entity
export const untransform = (model) => {
  const knexData = { extra_data: {} }

  const mainKeys = {
    id: "id",
    internalNumber: "internal_number",
    pvNumber: "pv_number",
    examinationDate: "examination_date",
    profile: "profile",
    askerId: "asker_id",
    addedBy: "added_by",
    hospitalId: "hospital_id",
  }

  Object.keys(model).forEach((key) => {
    if (mainKeys[key]) {
      knexData[mainKeys[key]] = model[key]
    } else {
      knexData.extra_data[key] = model[key]
    }
  })

  return knexData
}
