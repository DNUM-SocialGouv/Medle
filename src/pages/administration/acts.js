import { PropTypes } from "prop-types"
import React from "react"
import { Container } from "reactstrap"

import Layout from "../../components/Layout"
import { Title1 } from "../../components/StyledComponents"
import { withAuthentication } from "../../utils/auth"
import { ADMIN } from "../../utils/roles"

const ActsPage = ({ currentUser }) => {
  return (
    <Layout page="acts" currentUser={currentUser} admin={true}>
      <Container
        style={{ maxWidth: 980, minWidth: 740 }}
        className="mt-5 mb-5 d-flex justify-content-between align-items-baseline"
      >
        <Title1 className="">{"Administration des actes"}</Title1>
      </Container>
    </Layout>
  )
}

ActsPage.propTypes = {
  paginatedData: PropTypes.object,
  currentUser: PropTypes.object.isRequired,
}

export default withAuthentication(ActsPage, ADMIN)
