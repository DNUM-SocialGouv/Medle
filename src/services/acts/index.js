import { create } from "./create"
import { del } from "./delete"
import { exportActs } from "./export"
import { find } from "./find"
import { search } from "./search"
import { update } from "./update"

export { create, del, exportActs, find, search, update }
