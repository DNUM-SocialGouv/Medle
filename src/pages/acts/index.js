import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown"
import ArrowRightIcon from "@material-ui/icons/ArrowRight"
import ListAltIcon from "@material-ui/icons/ListAlt"
import Link from "next/link"
import PropTypes from "prop-types"
import React, { useEffect, useMemo, useState } from "react"
import { useForm } from "react-hook-form"
import Select from "react-select"
import AsyncSelect from "react-select/async"
import { Alert, Button, Col, Container, Form, FormGroup, Input, Label, Row, Table } from "reactstrap"

import { fetchExport, searchActs } from "../../clients/acts"
import { memoizedSearchAskers } from "../../clients/askers"
import { SearchButton } from "../../components/form/SearchButton"
import Layout from "../../components/Layout"
import Pagination from "../../components/Pagination"
import { Title1 } from "../../components/StyledComponents"
import { VerticalList } from "../../components/VerticalList"
import { isOpenFeature } from "../../config"
import { useDebounce } from "../../hooks/useDebounce"
import { usePaginatedData } from "../../hooks/usePaginatedData"
import { buildScope } from "../../services/scope"
import { profiles as profilesConstants } from "../../utils/actsConstants"
import { buildAuthHeaders, redirectIfUnauthorized, withAuthentication } from "../../utils/auth"
import { isoToFr } from "../../utils/date"
import { getReferenceData } from "../../utils/init"
import { logError } from "../../utils/logger"
import { ACT_CONSULTATION } from "../../utils/roles"
import { mapArrayForSelect } from "../../utils/select"

const ActsListPage = ({ paginatedData: initialPaginatedData, currentUser }) => {
  // const renderCount = React.useRef(0)
  // renderCount.current++

  const [paginatedData, error, loading, fetchPage] = usePaginatedData(searchActs, initialPaginatedData)
  const [isOpenedFilters, setOpenedFilters] = useState(false)
  const { register, unregister, handleSubmit, setValue, getValues } = useForm({})
  const [hospitals, setHospitals] = useState([])
  const [profiles, setProfiles] = useState([])
  const [asker, setAsker] = useState(null)
  const [search, setSearch] = useState("")
  useDebounce(onChange, 500, [search])
  const scope = useMemo(() => buildScope(currentUser), [currentUser])

  const hospitalsChoices = useMemo(
    () =>
      mapArrayForSelect(
        scope?.length === 0
          ? getReferenceData("hospitals")
          : getReferenceData("hospitals").filter((hospital) => scope.includes(hospital.id)),
        (elt) => elt.id,
        (elt) => elt.name
      ),
    [scope]
  )

  const existingProfiles = useMemo(
    () =>
      mapArrayForSelect(
        Object.keys(profilesConstants) || [],
        (elt) => elt,
        (elt) => elt
      ),
    []
  )

  useEffect(() => {
    // Extra field in form to store the value of selects
    register({ name: "hospitals" })
    register({ name: "profiles" })
    register({ name: "asker" })
    register({ name: "search" })

    return () => {
      unregister({ name: "hospitals" })
      unregister({ name: "profiles" })
      unregister({ name: "asker" })
      unregister({ name: "search" })
    }
  }, [register, unregister])

  const numFilters = Object.values(getValues()).filter((val) => !!val).length

  function toggleFilters() {
    setOpenedFilters((state) => !state)
    onChange()
  }

  function onHospitalsChange(selectedOption) {
    // Needs transformation between format of react-select to expected format for API call
    setValue("hospitals", !selectedOption?.length ? null : selectedOption)
    // Needs to sync specifically the value to the react-select as well
    setHospitals(selectedOption)
    onChange()
  }

  function onProfilesChange(selectedOption) {
    // Needs transformation between format of react-select to expected format for API call
    setValue("profiles", !selectedOption?.length ? null : selectedOption)

    // Needs to sync specifically the value to the react-select as well
    setProfiles(selectedOption)
    onChange()
  }

  function onAskerChange(selectedOption) {
    // Needs transformation between format of react-select to expected format for API call
    setValue("asker", selectedOption)

    // Needs to sync specifically the value to the react-select as well
    setAsker(selectedOption)
    onChange()
  }

  function onSearchChange(e) {
    const text = e?.target?.value || ""
    setSearch(text)
    setValue("search", text)
  }

  async function loadAskers(search) {
    const askers = await memoizedSearchAskers({ search })

    return mapArrayForSelect(
      askers?.elements,
      (elt) => elt.id,
      (elt) => elt.name + (elt.depCode ? ` (${elt.depCode})` : "")
    )
  }

  function onChange() {
    onSubmit(getValues())
  }

  function onSubmit(formData) {
    fetchPage(formData)(0)
  }

  async function onExport() {
    await fetchExport(getValues())
  }

  return (
    <Layout page="acts" currentUser={currentUser}>
      <Title1 className="mt-5 mb-4">{"Tous les actes"}</Title1>
      <Container style={{ maxWidth: 980 }}>
        <Form onSubmit={handleSubmit(onSubmit)}>
          <FormGroup inline className="mb-4 justify-content-center">
            <Row>
              <Col className="flex-grow-1">
                <Input
                  type="text"
                  name="search"
                  id="search"
                  placeholder="Rechercher un acte ou plusieurs, par n° interne ou n° de PV"
                  autoComplete="off"
                  value={search}
                  onChange={onSearchChange}
                />
              </Col>
            </Row>
            {isOpenFeature("export") && (
              <>
                <Row className="mt-3">
                  <Col>
                    <Button className="mr-3" color="secondary" outline={!isOpenedFilters} onClick={toggleFilters}>
                      Plus de critères {isOpenedFilters ? <ArrowDropDownIcon /> : <ArrowRightIcon />}
                    </Button>
                    {numFilters > 0 && (
                      <span style={{ color: "#6c757d" }}>
                        {numFilters} critère{numFilters > 1 && "s"} actif{numFilters > 1 && "s"}
                      </span>
                    )}
                  </Col>
                </Row>
                {isOpenFeature("export") && (
                  <div
                    className="p-3 mt-3 border rounded shadow-xl border-light bg-light"
                    style={{ display: isOpenedFilters ? "block" : "none" }}
                  >
                    <Row>
                      <Col sm="3">
                        <Label htmlFor="startDate" className="text-dark">
                          Date de début
                        </Label>
                        <Input
                          type="date"
                          name="startDate"
                          id="startDate"
                          placeholder="Date de début"
                          innerRef={register}
                          onChange={onChange}
                        />
                      </Col>
                      <Col sm="3">
                        <Label htmlFor="startDate" className="text-dark">
                          Date de fin
                        </Label>
                        <Input
                          type="date"
                          name="endDate"
                          id="endDate"
                          placeholder="Date de fin"
                          innerRef={register}
                          onChange={onChange}
                        />
                      </Col>
                    </Row>
                    {hospitalsChoices?.length > 1 && (
                      <Row className="mt-3">
                        <Col>
                          <Label className="text-dark">Établissements</Label>
                          <Select
                            options={hospitalsChoices}
                            value={hospitals}
                            isMulti
                            onChange={onHospitalsChange}
                            noOptionsMessage={() => "Aucun résultat"}
                            placeholder="Choisissez un établissement de votre périmètre"
                            isClearable={true}
                            isSearchable={true}
                          />
                        </Col>
                      </Row>
                    )}{" "}
                    <Row className="mt-3">
                      <Col>
                        <Label className="text-dark">Profils et autres activités</Label>
                        <Select
                          options={existingProfiles}
                          value={profiles}
                          isMulti
                          onChange={onProfilesChange}
                          noOptionsMessage={() => "Aucun résultat"}
                          placeholder="Choisissez un profil ou activité"
                          isClearable={true}
                          isSearchable={true}
                        />
                      </Col>
                    </Row>
                    <Row className="mt-3">
                      <Col>
                        <Label className="text-dark">Demandeur</Label>
                        <AsyncSelect
                          loadOptions={(search) => loadAskers(search)}
                          placeholder="Tapez le nom du demandeur"
                          noOptionsMessage={() => "Aucun résultat"}
                          loadingMessage={() => "Chargement..."}
                          onChange={onAskerChange}
                          isClearable={true}
                          isSearchable={true}
                          value={asker}
                        />
                      </Col>
                    </Row>
                  </div>
                )}
              </>
            )}
          </FormGroup>
        </Form>
        {error && (
          <Alert color="danger" className="mb-4">
            {error}
          </Alert>
        )}
        {!error && !paginatedData?.elements?.length && <div className="text-center">{"Aucun acte trouvé."}</div>}

        {!error && !!paginatedData?.elements?.length && (
          <>
            <div className="my-4 d-flex justify-content-center">
              <b>{paginatedData.totalCount}</b>&nbsp;résultat{paginatedData.totalCount > 1 && "s"}
            </div>
            <Table responsive className="table-hover">
              <thead>
                <tr className="table-light">
                  <th>N° dossier interne</th>
                  <th>N° PV</th>
                  <th>Date</th>
                  <th>Type de profil</th>
                  <th>{"Type d'acte"}</th>
                  <th />
                </tr>
              </thead>
              <tbody>
                {paginatedData.elements.map((act) => (
                  <Link key={act.id} href="/acts/[id]" as={`/acts/${act.id}`}>
                    <tr key={act.id}>
                      <td>
                        <b>{act.internalNumber}</b>
                      </td>
                      <td>{act.pvNumber}</td>
                      <td>{act.examinationDate && isoToFr(act.examinationDate)}</td>
                      <td>{act.profile}</td>
                      <td>{act.examinationTypes && <VerticalList content={act.examinationTypes} />}</td>
                      <td className="text-decoration">
                        <Link href="/acts/[id]" as={`/acts/${act.id}`}>
                          <a className="text-decoration-none">Voir</a>
                        </Link>
                      </td>
                    </tr>
                  </Link>
                ))}
              </tbody>
            </Table>
            <Pagination data={paginatedData} fn={fetchPage(getValues())} />
            {isOpenFeature("export") && (
              <div className="mt-5 d-flex justify-content-center">
                <SearchButton className="btn-outline-primary" disabled={loading} onClick={onExport}>
                  <ListAltIcon /> Exporter les données
                </SearchButton>
              </div>
            )}
          </>
        )}
      </Container>
    </Layout>
  )
}

ActsListPage.getInitialProps = async (ctx) => {
  const headers = buildAuthHeaders(ctx)
  try {
    const paginatedData = await searchActs({ headers })
    return { paginatedData }
  } catch (error) {
    logError("APP error", error)

    redirectIfUnauthorized(error, ctx)
  }
  return {}
}

ActsListPage.propTypes = {
  currentUser: PropTypes.object.isRequired,
  paginatedData: PropTypes.object.isRequired,
}

export default withAuthentication(ActsListPage, ACT_CONSULTATION)
