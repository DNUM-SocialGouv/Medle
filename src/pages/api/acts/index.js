import Cors from "micro-cors"

import { STATUS_200_OK, METHOD_GET, METHOD_OPTIONS, METHOD_POST } from "../../../utils/http"
import { ACT_CONSULTATION, ACT_MANAGEMENT } from "../../../utils/roles"
import { sendAPIError, sendMethodNotAllowedError } from "../../../services/errorHelpers"
import { checkValidUserWithPrivilege } from "../../../utils/auth"

import { create, search } from "../../../services/acts"

const handler = async (req, res) => {
   res.setHeader("Content-Type", "application/json")

   try {
      switch (req.method) {
         case METHOD_GET: {
            const currentUser = checkValidUserWithPrivilege(ACT_CONSULTATION, req, res)

            const { totalCount, currentPage: requestedPage, maxPage, byPage: LIMIT, elements: acts } = await search(
               req.query,
               currentUser,
            )

            return res
               .status(STATUS_200_OK)
               .json({ totalCount, currentPage: requestedPage, maxPage, byPage: LIMIT, elements: acts })
         }
         case METHOD_POST: {
            const currentUser = checkValidUserWithPrivilege(ACT_MANAGEMENT, req, res)

            const id = await create(req.body, currentUser)

            return res.status(STATUS_200_OK).json({ id })
         }

         default:
            return sendMethodNotAllowedError(res)
      }
   } catch (error) {
      sendAPIError(error, res)
   }
}

const cors = Cors({
   allowMethods: [METHOD_GET, METHOD_OPTIONS, METHOD_POST],
})

export default cors(handler)
