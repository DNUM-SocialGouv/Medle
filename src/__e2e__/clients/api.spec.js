import { authenticate } from "../../clients/authentication"
import { createAct, searchActsByKey } from "../../clients/acts"

const headersActUserTours = () => authenticate("acte@tours.fr", "test")

describe("/acts", () => {
  it("should be possible to an act operator of Tours to add an act", async () => {
    const { headers } = await headersActUserTours()

    await createAct({
      act: {
        addedBy: 2,
        askerId: 8,
        examinationDate: "2020-04-14",
        examinations: ["Autres"],
        examinationTypes: ["Somatique", "Psychiatrique"],
        hospitalId: 1,
        internalNumber: "wxwxwx123123",
        location: "UMJ",
        periodOfDay: "Nuit profonde",
        personAgeTag: "0-2 ans",
        personGender: "FÃ©minin",
        profile: "Victime (vivante)",
        pvNumber: "",
        violenceContexts: ["Autre type/Autre", "Infra-familiale (hors conjugale)"],
        violenceNatures: ["Sexuelle", "Attentat/Hyper Cacher"],
      },
      headers,
    })

    const acts = await searchActsByKey({ key: "internalNumber", value: "wxwxwx123123", headers })

    expect(acts[0]).toMatchSnapshot({
      id: expect.any(Number),
    })
  })
})
