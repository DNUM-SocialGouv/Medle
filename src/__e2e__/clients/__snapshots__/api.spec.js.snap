// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/acts should be possible to an act operator of Tours to add an act 1`] = `
Object {
  "asker": Object {
    "id": 8,
  },
  "examinationDate": "2020-04-14",
  "examinationTypes": Array [
    "Somatique",
    "Psychiatrique",
  ],
  "examinations": Array [
    "Autres",
  ],
  "hospital": Object {
    "id": 1,
  },
  "id": Any<Number>,
  "internalNumber": "wxwxwx123123",
  "location": "UMJ",
  "periodOfDay": "Nuit profonde",
  "personAgeTag": "0-2 ans",
  "personGender": "FÃ©minin",
  "profile": "Victime (vivante)",
  "pvNumber": "",
  "user": Object {
    "id": 2,
  },
  "violenceContexts": Array [
    "Autre type/Autre",
    "Infra-familiale (hors conjugale)",
  ],
  "violenceNatures": Array [
    "Sexuelle",
    "Attentat/Hyper Cacher",
  ],
}
`;
